creme/optim/schedulers.py

```diff
@@ -33,6 +34,35 @@ def get(self, t):
         return self.learning_rate
 
 
+class Cyclic(Scheduler):
+    """Cyclic learning schedule, in each cycle we linearly decrease the
+    learning rate from lr1 to lr2.
+
+    Parameters:
+        lr1 (float): Starting learning rate.
+        lr2 (float): Ending learning rate.
+        c (int): Cycle length.
+
+    References:
+        1. `Averaging Weights Leads to Wider Optima and Better Generalization
+            <https://arxiv.org/abs/1803.05407>`_
+
+    """
+
+    def __init__(self, lr1, lr2, c):
+        assert lr1 >= lr2, "lr1 should be greater than or equal to lr2"
```

Comment:
There shouldn't be `assert`s in library code! They're reserved for test code :). You should use `if`s and raise `ValueError` exceptions.

Reply:
Fixed it.